### PR TITLE
feat(server): check Response headers for Connection: close in keep_alive loop

### PR DESCRIPTION
**Pull request by seanmonstar**
BREAKING CHANGE: Usage of Response.deconstruct() and construct() now use
  a &mut Headers, instead of the struct proper.

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -204,11 +204,27 @@ where S: NetworkStream + Clone, H: Handler {
         }
 
         keep_alive = http::should_keep_alive(req.version, &req.headers);
-        let mut res = Response::new(&mut wrt);
-        res.version = req.version;
-        handler.handle(req, res);
-        debug!("keep_alive = {:?}", keep_alive);
+        let version = req.version;
```
Comment (reem) [inline src/server/mod.rs]: This function is starting to get *really* long, can you split this section into another fn?

**Comment by seanmonstar** [comment]
@reem ok, refactored that function...

**Comment by reem** [comment]
LGTM - do we care about the coveralls regression as a blocker?

**Comment by seanmonstar** [comment]
I don't, as I still find the coverage report confusing and inaccurate.

**Pull request merged**